src/bootstrap/lib.rs

```diff
@@ -1757,3 +1773,17 @@ fn envify(s: &str) -> String {
         .flat_map(|c| c.to_uppercase())
         .collect()
 }
+
+#[test]
+fn test_beta_rev_parsing() {
+    // single digit revision
+    assert_eq!(extract_beta_rev("1.99.9-beta.7"), Some("7"));
+    // multiple digits
+    assert_eq!(extract_beta_rev("1.99.9-beta.777"), Some("777"));
+    // nightly channel (no beta revision)
+    assert_eq!(extract_beta_rev("1.99.9-nightly"), None);
+    // stable channel (no beta revision)
+    assert_eq!(extract_beta_rev("1.99.9"), None);
+    // invalid string
+    assert_eq!(extract_beta_rev("invalid"), None);
+}
```

Comment:
Can you move this to above of https://github.com/rust-lang/rust/blob/2bd35159fa0321c9614ce50d585b376a90a6ae5f/src/bootstrap/builder/tests.rs#L149 line?

Reply:
Understood. Moved the test cases to the `bootstrap/builder/tests.rs` file.